subsys/net/lib/download_client/src/download_client.c

```diff
@@ -540,12 +551,23 @@ void download_thread(void *client, void *a, void *b)
 
 		rc = get_request_send(dl);
 		if (rc) {
+			int err;
+
+			if (errno == EHOSTDOWN) {
+				continue;
+			}
+
 			rc = error_evt_send(dl, ECONNRESET);
 			if (rc) {
 				/* Restart and suspend */
 				break;
 			}
-			reconnect(dl);
+			err = reconnect(dl);
+			if (err == ENETUNREACH) {
+				LOG_WRN("Network is down, "
+					"killing download thread");
+				return;
```

Comment:
Here we should also suspend the thread instead of killing it. And notify an event.

Reply:
Though I had updated, that text, fixing.